docs/mvc/tutorials/mvc-with-entity-framework.rst

```diff
@@ -1,8 +1,357 @@
+    :language: aspx-cs
+
+Run the app and click the "Books" link in the top nav bar. You should see a list of books. The links for create, edit, details, and delete are not functioning yet. We’ll add those next.
+
+Add a details page
+------------------
+
+Add the following method to the ``BooksController`` class:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/BookController.cs
+    :language: c#
+    :lines: 29-40
+    :dedent: 8
+
+This code looks up a book by ID. In the EF query:
+
+- The ``Include`` method tells EF to fetch the related ``Author`` entity.  
+- The ``SingleOrDefault`` method returns a single entity by ID, or ``null`` if none is found.
+
+If the EF query returns ``null``, the controller method returns ``HttpNotFound``, which ASP.NET translates into a 404 response. Otherwise, the controller passes book to a view, which renders the details page. Let’s add the view now.
+
+In the *Views/Book* folder, add a view named *Details.cshtml* with the following code:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Views/Book/Details.cshtml
+    :language: aspx-cs
+
+Add a create page
+-----------------
+
+Add the following two methods to ``BookController``:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/BookController.cs
+    :language: c#
+    :lines: 41-46,137-150
+    :dedent: 8
+
+Add a view named *Views/Book/Create.cshtml*.
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Views/Book/Create.cshtml
+    :language: aspx-cs
+
+This view renders an HTML form. In the ``form`` element, the ``asp-action`` tag helper specifies the controller action to invoke when the client submits the form. Notice that the form uses HTTP POST.
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Views/Book/Create.cshtml
+    :language: aspx-cs
+    :lines: 4
+    :dedent: 2
+
+Now let’s write the controller action to handle the form post. In the ``BookController`` class, add the following method.
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/BookController.cs
+    :language: c#
+    :lines: 50-66
+    :dedent: 8
+
+The ``[HttpPost]`` attribute tells MVC 6 that this action applies to HTTP POST requests. The ``[ValidateAntiForgeryToken]`` attribute is a security feature that guards against cross-site request forgery. For more information, see :doc:`../../security/anti-request-forgery`.
+
+Inside this method, we check the model state (``ModelState.IsValid``). If the client submitted a valid model, we add it to the database. Otherwise, we return the original view. This time, the view will display the validation errors in these ``span`` elements:
+
+.. literalinclude:: mvc-with-entity-framework/sample/src/ContosoBooks/Views/Book/Create.cshtml
+    :language: aspx-cs
+    :lines: 9
+    :dedent: 6
+
+Add validation rules to the book model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
I'm fine either way... just wanted to call out that the title of this particular walkthrough is "Create a web app that talks to a SQL database" and this section has nothing to do with that.